torchtitan/config_manager.py

```
@@ -638,6 +652,15 @@ def parse_args(self, args_list: list = sys.argv[1:]):
                 exp["pipeline_parallel_split_points"]
             )
 
+        if (
+            "experimental" in args_dict
+            and "model_module_path" in args_dict["experimental"]
+            and args_dict["experimental"]["model_module_path"]
+        ):
+            from torchtitan.models import add_model_spec_path
+
+            add_model_spec_path(args_dict["experimental"]["model_module_path"])
```

may I ask why putting it here, instead of in `build_model_specs`? Is it because you think it's better to fail early? I think in general in torchtitan, we are following the idea that we try to put fail check close to where something is being used, whose main benefit is that the checking and functioning are less scattered.